pkg/kubelet/kuberuntime/kuberuntime_sandbox.go

```diff
@@ -96,11 +96,14 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxConfig(pod *v1.Pod, attemp
 
 	if !kubecontainer.IsHostNetworkPod(pod) {
 		// TODO: Add domain support in new runtime interface
-		hostname, _, err := m.runtimeHelper.GeneratePodHostNameAndDomain(pod)
+		hostname, hostDomain, err := m.runtimeHelper.GeneratePodHostNameAndDomain(pod)
+		if err != nil {
+			return nil, err
+		}
+		podSandboxConfig.Hostname, err = m.runtimeHelper.GetHostNameForKernel(hostname, hostDomain, pod.Spec.HostnameFQDN)
```

Comment:
It seems podSandboxConfig.Hostname shouldn't be assigned if err is not nil

Reply:
agree, changing